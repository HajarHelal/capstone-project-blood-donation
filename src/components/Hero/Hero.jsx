import React from 'react';
import propTypes from 'prop-types';

function Hero({ img, children }) {
  return (
    <div
      className="flex w-100  h-screen bg-no-repeat bg-cover"
      style={{
        backgroundImage: `linear-gradient(rgba(0, 0, 0, 0.6),rgba(0, 0, 0, 0.6)),url(${img})`,
      }}
    >
      {children}
    </div>
  );
}
Hero.propTypes = {
  img: propTypes.string,
  children: propTypes.element.isRequired,
};

export default Hero;
