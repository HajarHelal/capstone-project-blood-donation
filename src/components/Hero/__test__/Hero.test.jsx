import React from 'react';
import renderer from 'react-test-renderer';
import Hero from '../Hero';
import HeroImg from '../images/Man-Smiling-Donating-Blood.jpg';

it('renders correctly', () => {
  const tree = renderer.create(<Hero img={HeroImg} />).toJSON();
  expect(tree).toMatchSnapshot();
});
