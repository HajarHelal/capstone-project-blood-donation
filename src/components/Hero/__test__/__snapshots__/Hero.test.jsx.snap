// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex w-100  h-screen bg-no-repeat bg-cover"
  style={
    Object {
      "backgroundImage": "linear-gradient(rgba(0, 0, 0, 0.6),rgba(0, 0, 0, 0.6)),url(Man-Smiling-Donating-Blood.jpg)",
    }
  }
/>
`;
